src/common/redis_module/ray_redis_module.cc

```diff
@@ -49,6 +49,26 @@
     return RedisModule_ReplyWithError(ctx, (MESSAGE)); \
   }
 
+static const char *table_prefixes[] = {
+    NULL, "TASK:", "CLIENT:", "OBJECT:", "FUNCTION:",
+};
+
+/// Format a pubsub channel for a specific key. pubsub_channel_str should
+/// contain a valid TablePubsub.
+RedisModuleString *FormatPubsubChannel(RedisModuleCtx *ctx,
+                                       RedisModuleString *pubsub_channel_str,
+                                       RedisModuleString *id) {
+  long long pubsub_channel_long;
+  RAY_CHECK(RedisModule_StringToLongLong(
+                pubsub_channel_str, &pubsub_channel_long) == REDISMODULE_OK)
+      << "Prefix must be a valid TablePrefix";
+  char pubsub_channel[sizeof(TablePubsub) + 1];
```

Comment:
Why +1?

Reply:
For the null terminator, although this doesn't measure the maximum number of characters needed properly...I changed this to `TablePubsub_MAX`.